### PR TITLE
Toptalkers

### PR DESCRIPTION
**Pull request by lgandras**
TopTalkers, comes together with all other VPC FlowLogs tools. Ideally we'd merge this before the 26th of April since there's a blog post coming.

### CODE REVIEW

**Comment by ludoo** [comment]
Andras, can you merge `main` again and fix the following lint errors? I'll then review as agreed

```
Step #0 - "lint": 
Step #0 - "lint": In ./tools/vpc-flowlogs-toptalkers/update-ip-range-labels.sh line 5:
Step #0 - "lint": DIR=$(dirname $0)
Step #0 - "lint":               ^-- SC2086: Double quote to prevent globbing and word splitting.
Step #0 - "lint": 
Step #0 - "lint": 
Step #0 - "lint": In ./tools/vpc-flowlogs-toptalkers/update-ip-range-labels.sh line 23:
Step #0 - "lint": source "${DIR}/ip-range-labels/venv/bin/activate"
Step #0 - "lint": ^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
Step #0 - "lint": 
Step #0 - "lint": Some files need to be formatted in ./tools/vpc-flowlogs-toptalkers - FAIL
Step #0 - "lint": ./tools/vpc-flowlogs-toptalkers/update-ip-range-labels.sh 
```

**Comment by lgandras** [comment]
This time I ran `cloud-build-local` (first learned of it :) and the job went through.

File: tools/vpc-flowlogs-toptalkers/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+### Google API IP address ranges
+
+This project comes with the pre-populated list of IP addresses for Google APIs. This IPs change from time to time, so there is a script `update-ip-range-labels.sh` which takes care of updating the list in `ip-range-labels.yaml`. You need to run it manually to update the list, it is not a part of Terraform run.
+
+### Labelling the traffic
+
+This solution allows adding custome labels to the specific IP ranges and ports. Edit corresponding sections of the `labels.yaml` to add the mapping between the hosts or subnets and text labels.
+
+To label the traffic on the specific port add it to the under the `port_labels` key.
```
Comment (ludoo) [inline tools/vpc-flowlogs-toptalkers/README.md:41]: "add it under"

File: tools/vpc-flowlogs-toptalkers/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - `ipv6_ranges_to_exclude`
+    - `ipv6_aggregate_prefix` - if subnet is not mentioned in the `labels.yaml` the resulting label will be `netaddr6-IPV6PREFIX`
+
+### Time ranges
+
+Please note that the `current_month_*` and `last_month_*` reports will process only the tables from the corresponding time ranges. If you need historical time ranges - please change the view implementation.
+
+## Usage
+
+Once installed with the right configuration values, you'll see several views with the names `top_talkers_report_{current|previous}_month_{daily|weekly|monthly}` under the newly created dataset. This dataset will automatically get populated by Cloud Operations with the VPC flow logs that are enabled in the project where the log sink resides. It may take some minutes for first entries to appear in the dataset.
```
Comment (ludoo) [inline tools/vpc-flowlogs-toptalkers/README.md:65]: "for the first entries"

File: tools/vpc-flowlogs-toptalkers/ip-range-labels/main.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+import json
```
Comment (ludoo) [inline tools/vpc-flowlogs-toptalkers/ip-range-labels/main.py:3]: Can you reorder lexically?

File: tools/vpc-flowlogs-toptalkers/ip-range-labels/main.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import netaddr
+import urllib.request
+import sys
+import yaml
+
+goog_url="https://www.gstatic.com/ipranges/goog.json"
```
Comment (ludoo) [inline tools/vpc-flowlogs-toptalkers/ip-range-labels/main.py:9]: please put constants in uppercase, and always add spaces around the `=` sign

Comment (ludoo) [reply tools/vpc-flowlogs-toptalkers/ip-range-labels/main.py:9]: please run a linter/formatter or manually add spaces around the `=` sign everywhere, except when passing named function args

**Review by ludoo — CHANGES_REQUESTED** [review]
A few minor points, looks good otherwise

**Review by ludoo — APPROVED** [review]

**Comment by ludoo** [comment]
There are still errors in the lint test

```
Step #0 - "lint": Testing formatting for python2 files in ./tools/vpc-flowlogs-toptalkers
Step #0 - "lint": Testing formatting for python3 files in ./tools/vpc-flowlogs-toptalkers
Step #0 - "lint": make: *** [Makefile:28: test] Error 1
Step #0 - "lint": Some files need to be formatted in ./tools/vpc-flowlogs-toptalkers - FAIL
Step #0 - "lint": ./tools/vpc-flowlogs-toptalkers/ip-range-labels/main.py
```

**Pull request merged**